.github/workflows/master_ci.yml

```diff
@@ -13,104 +13,153 @@ jobs:
-                    { os: ubuntu-20.04, cpp: g++-10, cc: gcc-10, install_cmd: g++-10 gcc-10 },
-                    { os: ubuntu-22.04, cpp: g++-11, cc: gcc-11, install_cmd: g++-11 gcc-11 },
-                    { os: ubuntu-22.04, cpp: g++-12, cc: gcc-12, install_cmd: g++-12 gcc-12 },
-                    { os: ubuntu-22.04, cpp: clang++, cc: clang, install_cmd: clang }  ]
+        compiler: [ { os: ubuntu-22.04, cpp: g++-11, cc: gcc-11, install_cmd: g++-11 gcc-11 },
+                    { os: ubuntu-24.04, cpp: g++-14, cc: gcc-14, install_cmd: g++-14 gcc-14 },
+                    { os: ubuntu-24.04, cpp: clang++, cc: clang, install_cmd: clang }  ]
 
     runs-on: ${{ matrix.compiler.os }}
     env:
-      BOOST_ROOT : "/usr/local/share/boost/1.72.0"
       CI_ACTIVE : 1
 
     steps:
       - name: Checkout HEAD
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 1
+        uses: actions/checkout@v4
 
       - name: Install Dependencies
         run: |
           sudo apt-get install libtbb-dev libboost-program-options-dev libhwloc-dev lcov gcovr ${{ matrix.compiler.install_cmd }}
 
-      - name: Install Mt-KaHyPar Test Suite
+      - name: Install Mt-KaHyPar
+        env:
+          CC: ${{ matrix.compiler.cc }}
+          CXX: ${{ matrix.compiler.cpp }}
+        run: |
+          rm -rf build
+          mkdir build
+          cd build
+          cmake .. --preset=default -DKAHYPAR_CI_BUILD=ON
+          make -j2 MtKaHyPar
+
+  mt_kahypar_static_build:
+    name: Statically Linked Build
+    strategy:
+      matrix:
+        compiler: [ { os: ubuntu-24.04, cpp: g++-14, cc: gcc-14, install_cmd: g++-14 gcc-14 },
+                    { os: ubuntu-24.04, cpp: clang++, cc: clang, install_cmd: clang }  ]
+
+    runs-on: ${{ matrix.compiler.os }}
+    env:
+      CI_ACTIVE : 1
+
+    steps:
+      - name: Checkout HEAD
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get install libtbb-dev libboost-program-options-dev libhwloc-dev libudev-dev lcov gcovr ${{ matrix.compiler.install_cmd }}
+
+      - name: Install Mt-KaHyPar
         env:
           CC: ${{ matrix.compiler.cc }}
           CXX: ${{ matrix.compiler.cpp }}
         run: |
-          git submodule update --init --recursive
           rm -rf build
           mkdir build
           cd build
-          cmake .. -DCMAKE_BUILD_TYPE=RELEASE -DKAHYPAR_CI_BUILD=ON
+          cmake .. --preset=minimal -DKAHYPAR_CI_BUILD=ON -DBUILD_SHARED_LIBS=OFF -DKAHYPAR_STATIC_LINK_DEPENDENCIES=ON
+          make -j2 MtKaHyPar
+          if [[ $(ldd mt-kahypar/application/MtKaHyPar | grep hwloc) ]]; then
+            echo "Error: hwloc is dynamically linked"
+            exit 1
+          fi
+          if [[ $(ldd mt-kahypar/application/MtKaHyPar | grep boost) ]]; then
+            echo "Error: boost is dynamically linked"
+            exit 1
+          fi
+
+  mt_kahypar_64bit_test:
```

Comment:
This tests if can compile with 64-bits IDs but does not test anything, and therefore is no guarantee that our 64-bit build works. Therefore, I would not waste CI time for it.